### PR TITLE
Howto write stream connector

### PR DESCRIPTION
**Pull request by bouda1**
I need someone who writes a better english than I !

### CODE REVIEW

File: doc/fr/developper/writestreamconnector.rst [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+.. note::
+   In a near future, this directory will be known by Lua scripts launched by broker.
+
+Write all information into a file
+=================================
+
+Store raw data
+**************
+
+We will store the **bbbdo2file.lua** LUA connector into the
```
Comment (kduret) [inline doc/fr/developper/writestreamconnector.rst]: too much "b" in bbbdo2file.lua ?

File: doc/fr/developper/writestreamconnector.rst [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+Centreon Broker has a log function to write logs into a file. We will use this 
+function to write raw data into a dedicated file. `See technical documentation for more information 
+<https://documentation.centreon.com/docs/centreon-broker/en/latest/exploit/stream_connectors.html#the-broker-log-object>`_.
+
+The **write()** function is called each time an event is received from a poller 
+through the broker. We will call the log function into this function to store 
+raw data. `See technical documentation for more information 
+<https://documentation.centreon.com/docs/centreon-broker/en/latest/exploit/stream_connectors.html#the-write-function>`_.
+
+The **bbbdo2file.lua** script will contain:
```
Comment (kduret) [inline doc/fr/developper/writestreamconnector.rst]: too much "b" in bbbdo2file.lua ?

File: doc/fr/developper/writestreamconnector.rst [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+      broker_log:info(3, k .. " => " .. tostring(v))
+    end
+    return true
+  end
+
+.. note::
+   For more information about the initialisation of the log function of Centreon
+   Broker and the parameters, please `see technical documentation <https://documentation.centreon.com/docs/centreon-broker/en/latest/exploit/stream_connectors.html#the-broker-log-object>`_.
+
+Once your file **/usr/share/centreon-broker/lua/bbbdo2file.lua** is ready, add
```
Comment (kduret) [inline doc/fr/developper/writestreamconnector.rst]: too much "b" in bbbdo2file.lua ?

File: doc/fr/developper/writestreamconnector.rst [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+  end
+
+.. note::
+   For more information about the initialisation of the log function of Centreon
+   Broker and the parameters, please `see technical documentation <https://documentation.centreon.com/docs/centreon-broker/en/latest/exploit/stream_connectors.html#the-broker-log-object>`_.
+
+Once your file **/usr/share/centreon-broker/lua/bbbdo2file.lua** is ready, add 
+execution rights on it::
+
+  # chown centreon-engine:centreon-engine /usr/share/centreon-broker/lua/bbbdo2file.lua
```
Comment (kduret) [inline doc/fr/developper/writestreamconnector.rst]: too much "b" in bbbdo2file.lua ?

File: doc/fr/developper/writestreamconnector.rst [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+
+.. image:: /_static/images/developper/lua/visualize_data_grafana.png
+   :align: center
+   :scale: 65%
+
+Discover other Centreon Stream Connectors
+*****************************************
+
+Centreon provides a Github repository to host LUA scripts developed by Centreon 
+and the community. Please go to the `dedicated Github <http://github.com/centreon/centreon-lua-scripts>`_.
```
Comment (kduret) [inline doc/fr/developper/writestreamconnector.rst]: github repository is : https://github.com/centreon/centreon-stream-connector-scripts

**Comment by bouda1** [comment]
Thanks @kduret , I forgot to push the french version. Now, we have a french and an english versions, ... in english.

File: doc/en/developer/writestreamconnector.rst [hunk 6]
```diff
@@ -0,0 +1,898 @@
...
+    influx_database             = "mydb",
+    influx_user                 = "",
+    influx_password             = "",
+    max_buffer_size             = 5000,
+    max_buffer_age              = 5
+  }
+
+
+In this table, we give default values to parameters that can be possibly
+changed during the **init()** call. This table will be use to store important
```
Comment (thiuyendang) [inline doc/en/developer/writestreamconnector.rst]: used

File: doc/en/developer/writestreamconnector.rst [hunk 7]
```diff
@@ -0,0 +1,898 @@
...
+  SERVICE:mail-jupiter-frontend;85;memory-stats;495;0;Memory usage (Total 12.0GB): 0.57GB [buffer:0.04GB] [cache:0.23GB] [pages_tables:0.02GB] [mapped:0.02GB] [active:0.07GB] [inactive:0.03GB] [apps:0.16GB] [unused:11.43GB]
+  SERVICE:mail-io-backend;88;traffic-eth1;547;0;Traffic In : 1.51 Mb/s (1.51 %), Out : 7.12 Mb/s (7.12 %) - Total RX Bits In : 389.61 Gb, Out : 390.54 Gb
+  SERVICE:mail-io-backend;88;diskio-system;551;0;Device /dev/sda: avg read 4.78 (MB/s) and write 9.08 (MB/s)
+
+
+***********************************
+Export performance data to InfluxDB
+***********************************
+
+Now, you have already seen many things on stream connectors. It is time to
```
Comment (thiuyendang) [inline doc/en/developer/writestreamconnector.rst]: ...things about stream connectors.

File: doc/en/developer/writestreamconnector.rst [hunk 8]
```diff
@@ -0,0 +1,898 @@
...
+    if category == 3 and element == 1 then
+      return true
+    end
+    return false
+  end
+
+Complete script
+===============
+
+The complete script can be download `here <https://github.com/centreon/centreon-stream-connector-scripts/tree/master/influxdb>`_.
```
Comment (thiuyendang) [inline doc/en/developer/writestreamconnector.rst]: downloaded

File: doc/fr/developper/writestreamconnector.rst [hunk 9]
```diff
@@ -0,0 +1,898 @@
...
+
+  function write(d)
+    for k,v in pairs(d) do
+      broker_log:info(3, k .. " => " .. tostring(v))
+    end
+    return true
+  end
+
+.. note::
+   Informations about the initialisation of the Broker's log function
```
Comment (thiuyendang) [inline doc/fr/developper/writestreamconnector.rst]: initialization

File: doc/fr/developper/writestreamconnector.rst [hunk 10]
```diff
@@ -0,0 +1,898 @@
...
+  SERVICE:mail-jupiter-frontend;85;memory-stats;495;0;Memory usage (Total 12.0GB): 0.57GB [buffer:0.04GB] [cache:0.23GB] [pages_tables:0.02GB] [mapped:0.02GB] [active:0.07GB] [inactive:0.03GB] [apps:0.16GB] [unused:11.43GB]
+  SERVICE:mail-io-backend;88;traffic-eth1;547;0;Traffic In : 1.51 Mb/s (1.51 %), Out : 7.12 Mb/s (7.12 %) - Total RX Bits In : 389.61 Gb, Out : 390.54 Gb
+  SERVICE:mail-io-backend;88;diskio-system;551;0;Device /dev/sda: avg read 4.78 (MB/s) and write 9.08 (MB/s)
+
+
+***********************************
+Export performance data to InfluxDB
+***********************************
+
+Now, you have already seen many things on stream connectors. It is time to
```
Comment (thiuyendang) [inline doc/fr/developper/writestreamconnector.rst]: about stream connectors.

File: doc/fr/developper/writestreamconnector.rst [hunk 11]
```diff
@@ -0,0 +1,898 @@
...
+    influx_database             = "mydb",
+    influx_user                 = "",
+    influx_password             = "",
+    max_buffer_size             = 5000,
+    max_buffer_age              = 5
+  }
+
+
+In this table, we give default values to parameters that can be possibly
+changed during the **init()** call. This table will be use to store important
```
Comment (thiuyendang) [inline doc/fr/developper/writestreamconnector.rst]: used

File: doc/fr/developper/writestreamconnector.rst [hunk 12]
```diff
@@ -0,0 +1,898 @@
...
+    if category == 3 and element == 1 then
+      return true
+    end
+    return false
+  end
+
+Complete script
+===============
+
+The complete script can be download `here <https://github.com/centreon/centreon-stream-connector-scripts/tree/master/influxdb>`_.
```
Comment (thiuyendang) [inline doc/fr/developper/writestreamconnector.rst]: downloaded

**Pull request closed without merging**